acme/acme.go

```diff
@@ -183,7 +184,8 @@ func (a *ACME) leadershipListener(elected bool) error {
 		account := object.(*Account)
 		account.Init()
 		// Reset Account values if caServer changed, thus registration URI can be updated
-		if account != nil && account.Registration != nil && !strings.HasPrefix(account.Registration.URI, a.CAServer) {
+		if account != nil && account.Registration != nil && isAccountMatchingCaServer(account.Registration.URI, a.CAServer) {
```

Comment:
It's should be `!isAccountMatchingCaServer(account.Registration.URI, a.CAServer)` instead of `isAccountMatchingCaServer(account.Registration.URI, a.CAServer)` no ?